kernel/src/config.rs

```diff
@@ -0,0 +1,802 @@
+            enable_change_data_feed,
+            bool,
+            false
+        ),
+        (
+            "true to enable deletion vectors and predictive I/O for updates.",
+            DeltaConfigKey::EnableDeletionVectors,
+            enable_deletion_vectors,
+            bool,
+            // in databricks the default is dependent on the workspace settings and runtime version
+            // https://learn.microsoft.com/en-us/azure/databricks/administration-guide/workspace-settings/deletion-vectors
+            false
+        ),
+        (
+            "The number of columns for Delta Lake to collect statistics about for data skipping.",
+            DeltaConfigKey::DataSkippingNumIndexedCols,
+            num_indexed_cols,
+            i32,
+            32
+        ),
+        (
+            "whether to cleanup expired logs",
+            DeltaConfigKey::EnableExpiredLogCleanup,
+            enable_expired_log_cleanup,
+            bool,
+            true
+        ),
+        (
+            "Interval (number of commits) after which a new checkpoint should be created",
+            DeltaConfigKey::CheckpointInterval,
+            checkpoint_interval,
+            i32,
+            10
+        ),
+    );
+
+    /// The shortest duration for Delta Lake to keep logically deleted data files before deleting
+    /// them physically. This is to prevent failures in stale readers after compactions or partition overwrites.
+    ///
+    /// This value should be large enough to ensure that:
+    ///
+    /// * It is larger than the longest possible duration of a job if you run VACUUM when there are
+    ///   concurrent readers or writers accessing the Delta table.
+    /// * If you run a streaming query that reads from the table, that query does not stop for longer
+    ///   than this value. Otherwise, the query may not be able to restart, as it must still read old files.
+    pub fn deleted_file_retention_duration(&self) -> Duration {
+        lazy_static! {
+            static ref DEFAULT_DURATION: Duration = parse_interval("interval 1 weeks").unwrap();
+        }
+        self.0
+            .get(DeltaConfigKey::DeletedFileRetentionDuration.as_ref())
+            .and_then(|v| parse_interval(v).ok())
+            .unwrap_or_else(|| DEFAULT_DURATION.to_owned())
+    }
+
+    /// How long the history for a Delta table is kept.
+    ///
+    /// Each time a checkpoint is written, Delta Lake automatically cleans up log entries older
+    /// than the retention interval. If you set this property to a large enough value, many log
+    /// entries are retained. This should not impact performance as operations against the log are
+    /// constant time. Operations on history are parallel but will become more expensive as the log size increases.
+    pub fn log_retention_duration(&self) -> Duration {
+        lazy_static! {
+            static ref DEFAULT_DURATION: Duration = parse_interval("interval 30 days").unwrap();
+        }
```

Comment:
will lazy static (or lazy lock) ensure this is only materialized if we hit the `unwrap_or_else`?